Lib/email/_header_value_parser.py

```diff
@@ -1513,16 +1513,20 @@ def get_obs_local_part(value):
                 raise
             token, value = get_cfws(value)
         obs_local_part.append(token)
-    if (obs_local_part[0].token_type == 'dot' or
+    if not obs_local_part:
+        obs_local_part.defects.append(errors.InvalidHeaderDefect(
+            "abandoned parse; truncated value?"))
+    else:
+        if (obs_local_part[0].token_type == 'dot' or
             obs_local_part[0].token_type=='cfws' and
             obs_local_part[1].token_type=='dot'):
```

Comment:
This can still fail if `len(obs_local_part) == 1 and obs_local_part[0].token_type=='cfws'`.

Can you add a corresponding test?